services/path.py

```diff
@@ -174,11 +178,16 @@ def __init__(
 
     @cached_property
     def files(self) -> List[str]:
+        files = []
+
         if self.filter_flags:
             files = report_service.files_belonging_to_flags(
                 commit_report=self.unfiltered_report, flags=self.filter_flags
             )
+            if self.filter_paths:
+                files = list(set(files) & set(self.report.files))
```

Comment:
why are we getting the intersection between files and report.files here 🤔? 

Some thoughts:
- We are using `self.unfiltered_report` to get files that belong to flags from sessions, we might need to do the same thing for filtered report. unless report.files returns the filtered files expected then we might as well just use that.
- if it's any easier, we might need to filter `report.files` by paths and just return the set of the combined array

Reply:
Discussed over Slack.